services/src/main/java/org/exoplatform/addons/gamification/rest/ChallengeRest.java

```diff
@@ -100,8 +100,11 @@ public Response getChallengeById(@ApiParam(value = "Challenge technical id", req
     String currentUserId = Utils.getCurrentUser();
     try {
       Challenge challenge = challengeService.getChallengeById(challengeId, currentUserId);
-      List<Announcement> announcementList = announcementService.findAllAnnouncementByChallenge(challengeId, offset, limit);
-      return Response.ok(EntityMapper.fromChallenge(challenge, announcementList)).build();
+      if (challenge != null) {
```

Comment:
I will be better if we do the opposite check and return a 404 error if the challenge is not found